src/subscriptions-register.jsx

```diff
@@ -75,13 +75,12 @@ class SubscriptionRegisterDialog extends React.Component {
             );
         }
         let insights;
-        let insights_checkbox_disabled = true;
-        if (this.props.insights_available === true) {
-            insights_checkbox_disabled = false;
-        } else {
-            if (this.props.auto_attach === true) {
-                insights_checkbox_disabled = false;
-            }
+        let insights_checkbox_disabled = false;
+        // Disable registration of insights-client only in the corner case, when
+        // insights-client is not installed, and it is not possible to install it,
+        // because PackageKit is not available
+        if (this.props.insights_available === false) {
+            insights_checkbox_disabled = true;
         }
```

Comment:
actually this block can be dropped altogether: in practice, after removing the `auto_attach` attribute, the code will always set `insights_checkbox_disabled` to `false`, and thus never disable the insights checkbox; previously, it was disabled only when not auto-attaching

Reply:
Are you sure? Because I already removed it, but you enforced to add it back ... Do you remember: https://github.com/candlepin/subscription-manager-cockpit/pull/68#discussion_r1647310800 ?